File: packages/rownd-nodejs/src/telemetry/axiomTelemetryClient.ts
import { PLUGIN_ID } from "../constants";
import { RowndTelemetryClient, RowndTelemetryEvent } from "../types";

const DEFAULT_AXIOM_URL = "https://api.axiom.co/v1/datasets";

export class AxiomTelemetryClient implements RowndTelemetryClient {
  private readonly url: string;

  constructor(
    private readonly token: string,
    private readonly dataset: string,
    endpoint?: string,
  ) {
    const baseUrl = (endpoint ?? DEFAULT_AXIOM_URL).replace(/\/$/, "");
    this.url = `${baseUrl}/${dataset}/ingest`;
  }

  async recordEvent(event: RowndTelemetryEvent): Promise<void> {
    await fetch(this.url, {
      method: "POST",
      headers: {
        Authorization: `Bearer ${this.token}`,
        "Content-Type": "application/json",
      },
      body: JSON.stringify([
        {
          ts: new Date().toISOString(),
          plugin: PLUGIN_ID,
          ...event,
        },
      ]),
    });
  }
}

File: packages/rownd-python/src/supertokens_rownd/telemetry/axiom_telemetry_client.py
from __future__ import annotations

from typing import Optional

import httpx

from ..types import JsonDict


class AxiomTelemetryClient:
    def __init__(self, token: str, dataset: str, url: Optional[str] = None):
        self.token = token
        self.dataset = dataset
        self.url = (url or "https://api.axiom.co/v1/datasets").rstrip("/")

    async def record_event(self, event: JsonDict) -> None:
        async with httpx.AsyncClient(timeout=5.0) as client:
            await client.post(
                "%s/%s/ingest" % (self.url, self.dataset),
                headers={"Authorization": "Bearer %s" % self.token},
                json=[event],
            )
